### PR TITLE
Fix for <fanartres> not being used in advancedsettings.xml

### PR DESCRIPTION
**Pull request by roidy**
Fix to address https://github.com/xbmc/xbmc/issues/17739

## Description

Fix for `<fanartres>` in advancedsettings.xml not being used.

## Motivation and Context

Fixes the issue listed here https://github.com/xbmc/xbmc/issues/17739

## How Has This Been Tested?

Built and tested on Windows 10 

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/pictures/Picture.cpp [hunk 1]
```diff
@@ -214,12 +214,12 @@ bool CPicture::CacheTexture(uint8_t *pixels, uint32_t width, uint32_t height, ui
   uint32_t max_height = advancedSettings->m_imageRes;
   if (advancedSettings->m_fanartRes > advancedSettings->m_imageRes)
   { // 16x9 images larger than the fanart res use that rather than the image res
-    if (fabsf(static_cast<float>(width) / static_cast<float>(height) / (16.0f / 9.0f) - 1.0f) <= 0.01f &&
-        height >= advancedSettings->m_fanartRes)
+    if (fabsf(static_cast<float>(width) / static_cast<float>(height) / (16.0f / 9.0f) - 1.0f) <= 0.01f)
```
Comment (Rechi) [inline xbmc/pictures/Picture.cpp:217]: > The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

File: xbmc/pictures/Picture.cpp [hunk 2]
```diff
@@ -214,12 +214,12 @@ bool CPicture::CacheTexture(uint8_t *pixels, uint32_t width, uint32_t height, ui
...
     {
-      max_height = advancedSettings->m_fanartRes;
+      max_height = advancedSettings->m_fanartRes; // use height defined in fanartRes
     }
   }
+
```
Comment (Rechi) [inline xbmc/pictures/Picture.cpp:222]: Remove the white space at the end of the line.

**Review by Rechi — COMMENTED** [review]
Please follow the [current code guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md).

File: xbmc/pictures/Picture.cpp (hunk 1 above)
Comment (roidy) [reply xbmc/pictures/Picture.cpp:217]: This file contains multiple line that are over 100 characters in length, my changes do not increase the length of any lines.

I'm unsure what you expect me to do?

Comment (phunkyfish) [reply xbmc/pictures/Picture.cpp:217]: Only the lines that you have changed need to follow the guidelines.

File: xbmc/pictures/Picture.cpp [hunk 3]
```diff
@@ -214,12 +214,13 @@ bool CPicture::CacheTexture(uint8_t *pixels, uint32_t width, uint32_t height, ui
   uint32_t max_height = advancedSettings->m_imageRes;
   if (advancedSettings->m_fanartRes > advancedSettings->m_imageRes)
   { // 16x9 images larger than the fanart res use that rather than the image res
-    if (fabsf(static_cast<float>(width) / static_cast<float>(height) / (16.0f / 9.0f) - 1.0f) <= 0.01f)
+    if (fabsf(static_cast<float>(width) / static_cast<float>(height) / (16.0f / 9.0f) - 1.0f)
```
Comment (phunkyfish) [inline xbmc/pictures/Picture.cpp:217]: I think you still have whitespace at the end of this line.

**Comment by phunkyfish** [comment]
Just make sure there is no whitespace at the end of the lines you have modified.

Also, you should squash this down to the original commit only. If you don't know how to squash it down just ask.

**Comment by roidy** [comment]
> Also, you should squash this down to the original commit only.

Could you please explain how to do this or point me to the information, thanks

**Comment by howie-f** [comment]
easiest would be on command line, now that you have 3 commits, you do:
`git reset --soft HEAD~3`
then `git commit -m "Fix for <fanartres> not... (your message)"`
finally `git push -f`

**Comment by roidy** [comment]
Thanks, I'll give that a try after the build finishes.

File: xbmc/pictures/Picture.cpp [hunk 4]
```diff
@@ -214,12 +214,13 @@ bool CPicture::CacheTexture(uint8_t *pixels, uint32_t width, uint32_t height, ui
   uint32_t max_height = advancedSettings->m_imageRes;
   if (advancedSettings->m_fanartRes > advancedSettings->m_imageRes)
   { // 16x9 images larger than the fanart res use that rather than the image res
-    if (fabsf(static_cast<float>(width) / static_cast<float>(height) / (16.0f / 9.0f) - 1.0f) <= 0.01f &&
-        height >= advancedSettings->m_fanartRes)
+    if (fabsf(static_cast<float>(width) / static_cast<float>(height) / (16.0f / 9.0f) - 1.0f)
+        <= 0.01f)
```
Comment (howie-f) [inline xbmc/pictures/Picture.cpp:218]: well done! imo the following suggestion by clang-format
```suggestion
    if (fabsf(static_cast<float>(width) / static_cast<float>(height) / (16.0f / 9.0f) - 1.0f) <=
        0.01f)
```
can be ignored. @phunkyfish can tell us :)

**Comment by phunkyfish** [comment]
You need git add before the commit.

Alternatively you can undo 2 commits and amend the first one:

```
git reset HEAD~2
git add -u
git commit —amend 
```

**Comment by howie-f** [comment]
@phunkyfish: reset --soft = no git add needed ;)

**Comment by phunkyfish** [comment]
Well you learn something new every day 😉

File: xbmc/pictures/Picture.cpp (hunk 4 above)
Comment (phunkyfish) [reply xbmc/pictures/Picture.cpp:218]: I think either is ok. But as you are changing it anyway by squashing it’s up to you 😉

Comment (howie-f) [reply xbmc/pictures/Picture.cpp:218]: it‘s already squashed 😀 @roidy we‘re good i think

**Review by phunkyfish — APPROVED** [review]
Nice

**Comment by phunkyfish** [comment]
Thanks for the contribution @roidy ;)

**Pull request merged**